.pre-commit-config.yaml

```diff
@@ -15,10 +15,23 @@ repos:
     hooks:
       - id: isort
         files: ^python/cucim/src/.*
-        args: ["--settings-path=python/cucim/setup.cfg"]
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
+        args: ["--settings-path=python/cucim/pyproject_.toml"]
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
     hooks:
-      - id: flake8
-        args: ["--config=python/cucim/setup.cfg"]
-        files: ^python/cucim/.*
+      - id: black
+        files: (python|legate)/.*
+        args: ["--config", "python/cucim/pyproject_.toml"]
+  - repo: https://github.com/astral-sh/ruff-pre-commit
```

Comment:
I won't advocate for changes in this PR due to its size, but for some potential follow-up work, you might want to update the linter versions (all RAPIDS repos' linters are independent, you don't have to use a specific isort/black/ruff version). Also you can use `ruff` in place of `isort` if you enable the `I` category, and you can enable auto-fixes as well, in `pyproject.toml`. Enabling auto-fixes plays very nicely with pre-commit.

```
[tool.ruff]
select = ["E", "F", "W", "I"]
fix = true
```

Reply:
Sure. I also discussed with @jakirkham and @gigony that we can add a `.git-blame-ignore-revs` in a follow-up so all these linting changes [don't show up in the blame view on GitHub](https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/).

Reply:
Greg opened a follow-up PR: https://github.com/rapidsai/cucim/pull/622

Please let us know if you have thoughts there 🙂